app/controllers/patients_controller.rb

```diff
@@ -1,20 +1,36 @@
 class PatientsController < ApplicationController
   before_action :set_patient, only: [:show, :edit, :update, :destroy]
-  before_action :check_access, only: [:show, :edit, :update, :destroy] 
-  before_action :check_read_only, only: [:edit, :update, :destroy] 
-  before_action :check_demo, only: [:index, :show, :edit, :update, :destroy] 
+  before_action :check_access, only: [:show, :edit, :update, :destroy]
+  before_action :check_read_only, only: [:edit, :update, :destroy]
+  before_action :check_demo, only: [:index, :show, :edit, :update, :destroy]
 
   # GET /patients
   # GET /patients.json
   def index
     user = current_user
+    group = user.groups
     sub_ids = []
+    @patients = []
     if not current_user.admin
       if params[:result] == "true"
-        @patients = user.patients.where(result: true).order(:case_id).page(params[:page]).per(25)
+        if not group.empty?
+          group.each do |g|
+            @patients += g.patients.where(result: true).order(:case_id)
+          end
+        else
+          @patients = user.patients.where(result: true).order(:case_id)
+        end
       else
-        @patients = user.patients.order(:case_id).page(params[:page]).per(25)
+        if not group.empty?
+          group.each do |g|
+            @patients += g.patients.order(:case_id)
+          end
+        else
+          @patients = user.patients.order(:case_id)
+        end
       end
+      @patients = @patients.uniq
+      @patients = Kaminari.paginate_array(@patients).page(params[:page]).per(25)
```

Comment:
why you need to use Kaminari.paginate_array? @patients.order(:case_id).page(params[:page]).per(25) should be enough. And is it in the order of case_id? if user has two groups and the case 12345 in group 1 and 123 in group 2, 12345 will be the first patient and 123 will be the second in the patient list.

Reply:
@patients is an array of active records. Array only has uniq , there is no distinct method for an array

Reply:
Kaminari.paginate_array --> since an array is not a paginatable object, we need the wrapper from the kaminari package to convert it to a paginatable array object. 
Cannot directly use .page(params[:page]).per(25) for an array.

As far as the ordering goes, currently, patients within each group are ordered according to caseid and patient index lists patients group wise.. Do we not want to list per group and instead just order by caseid, meaning the patients from group1 and group2 might be shuffeled in the list?